### ISSUE TITLE
Option to require explicit field discarding

### ISSUE THREAD

**Issue by oleg-py** [scalalandio/chimney#53]
I would very much appreciate if usage of default values could be turned off.

At my current project we use default values quite liberally. I use chimney to transform between semi-normalized model objects and huge fat DTOs for frontend to consume.

Combined with the fact that chimney implicitly removes extra fields, adding new fields tends to not be typo-safe:

```
  case class DTO(media: List[String] = List())
  case class Model(medias: List[String] = List())

  DTO(List("1", "2", "3")).transformInto[Model] // Gives B(Nil), silently losing data at runtime
```

Furthermore, I get that behavior if I add a field with default to model, but forget to add it to DTO. Since frontend might take a week to catch up, it's quite a pain to figure out after doing a couple other things 😅 

If possible, I would like to get a compile-time error with these cases.

Another possible alternative is the option to disable default values. Both would be nice to have, actually

**Comment by krzemin** [comment]
It would be rather hard to provide general support for typos in presence of default values. What is a typo for you, might be expected behavior for others.

Support for default values were added quite time ago. I agree it might not be something that all users want, so it'd be worth to add modifier that disables looking for default values, something like:

```
DTO(List("1", "2", "3")).into[Model].disbaleDefaultValues.transform
```

would then result with compile-time error.

**Comment by krzemin** [comment]
Will be released soon in 0.1.9
